packages/nimble/R/BUGS_modelDef.R

```diff
@@ -668,29 +668,42 @@ modelDefClass$methods(expandDistributions = function() {
 
 modelDefClass$methods(checkMultivarExpr = function() {
     checkForExpr <- function(expr) {
-        output <- FALSE
+        ##output <- FALSE
         if(length(expr) == 1 && class(expr) %in% c("name", "numeric")) return(FALSE)
-        if(!deparse(expr[[1]]) %in% c('[', ':')) return(TRUE)
-        for(i in 2:length(expr)) 
-            if(checkForExpr(expr[[i]])) output <- TRUE
-        return(output)
+        if(!deparse(expr[[1]]) == '[') return(TRUE)
+        ## recurse only on the first argument of the `[`
+        return(checkForExpr(expr[[2]]))
+        ## Previously we recursed more completely.  Now we stop because expressions
+        ## inside `[` are allowed.
+        ## if(!deparse(expr[[1]]) %in% c('[', ':')) return(TRUE)
+        ## for(i in 2:length(expr)) 
+        ##     if(checkForExpr(expr[[i]])) output <- TRUE
+        ## return(output)
     }
 
     for(i in seq_along(declInfo)) {
         BUGSdecl <- declInfo[[i]]
         if(BUGSdecl$type != 'stoch') next
-        dist <- deparse(BUGSdecl$valueExpr[[1]])
+        ## dist <- deparse(BUGSdecl$valueExpr[[1]])
+        dist <- BUGSdecl$distributionName
+        ## The following line is a one-time insertion to break testing in any case
+        ## where the condition fails.
+        ## If the condition is always met, we can use BUGSdecl$distributionName in place of deparse(BUGSdecl$valueExpr[[1]])
+        ## if(dist != BUGSdecl$distributionName)
+        ##     stop(paste0("dist (", dist,") != BUGSdecl$distributionName (",BUGSdecl$distributionName,")"))
         types <- nimble:::distributionsInputList[[dist]]$types
```

Comment:
@paciorek Here is one place we need to look up types and I am not sure this is the right way to do it.  I think we want more processed types (I can see that they exist somewhere) and also to be sure that types from user-defined distributions will be accessed.